### DISCUSSION TITLE
Make description fields for Pipeline Elements and Adapter collabsile

### DISCUSSION THREAD

**Discussion by bossenti** [apache/streampipes#1328] [Ideas]
Within #1306 we started a discussion about how and if we want provide more detailed descriptions than our usual one liner in the UI form when creating pipeline elements or adapters.
One idea is to represent the description for parameters as a collapsible UI element, so that we can offer more detailed explanations   optionally. Therefore, we would need to extend our `strings` file by an optional field `long_description` that would then be rendered to a collapsible UI item. For more context information have a look at: https://github.com/apache/streampipes/pull/1306#discussion_r1110143723

How do you think about this idea? Would this be possible to implement?

**Comment by tenthe** [comment]
We could either do this with another field or add a separator at the end of the description and continue with the long description.
This way it would be backwards compatible and we would have less properties in the `strings` file. But I am not sure if this is a better solution.

**Reply by bossenti** [reply]
I'm definitely in favor of building it in such way that it is backwards compatible.
Currently, in my head the `description-long` field would be optional in the `strings` file. So you can provide this feature for new pipeline elements but are not forced to. And if there is no such field then the UI simply does not offer to extend the description field.

**Comment by bossenti** [comment]
Is this a suitable topic for GSOC? 🤔

**Reply by dominikriemer** [reply]
I think that's a rather small enhancement...we need to do some design refactoring in the UI anyways as I'm currently struggling with many deprecated layout options in Angular Material...so I can take care of that afterwards.
